### PR TITLE
Add XMSS and some of its parameter sets.

### PR DESCRIPTION
**Pull request by dingens**
Includes some minor schema changes

### CODE REVIEW

File: schema/flavor.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
             "type": "string",
             "enum": [
                 "IND-CPA",
-                "IND-CCA"
+                "IND-CCA",
```
Comment (franziskuskiefer) [inline schema/flavor.json]: This is a signature scheme, how can it be IND-CCA secure?

File: schema/paramset.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
             "$comment": "0 for 0, log_2 else"
         },
         "failure probability comment": {
-            "type": "integer"
```
Comment (franziskuskiefer) [inline schema/paramset.json]: This is a change in the schema and shouldn't be in this PR.

File: schema/scheme.json [hunk 3]
```diff
@@ -85,6 +85,9 @@
                 ]
             }
         },
+        "problems comment": {
```
Comment (franziskuskiefer) [inline schema/scheme.json]: same

File: signatures/xmss/xmss-sha2/param/xmss-sha2-10-256.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    classical: 256
+    quantum: 256
+number of operations: 2^10
+failure probability: 0
+sizes:
+    sk: 3116
+    pk: 68
+    ct/sig: 2500
+    msg: 256
+    comment: "value for `sk` depends vastly on the implementation, and can be balanced against signing time. The given value is when using `treehash` from BDS08, estimated. Source: https://crypto.stackexchange.com/a/77292 . Absolute minimum is 132 bytes."
```
Comment (franziskuskiefer) [inline signatures/xmss/xmss-sha2/param/xmss-sha2-10-256.yaml]: The [original paper](https://eprint.iacr.org/2011/484.pdf) discusses some of the trade-offs around private key size. That's maybe a better reference.

File: signatures/xmss/xmss-sha2/param/xmss-sha2-10-256.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+name: XMSS-SHA2_10_256
+security level:
+    classical: 256
+    quantum: 256
```
Comment (franziskuskiefer) [inline signatures/xmss/xmss-sha2/param/xmss-sha2-10-256.yaml]: How can these be the same? XMSS 256 should be 128-bit quantum secure, right?

File: signatures/xmss/xmss.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  - Stefan-Lukas Gazdag
+  - Joost Rijneveld
+  - Aziz Mohaisen
+stateful: true
+nist round: 0
+problems: []
+problems comment: "Only properties of hash functions assumed. This is implicitly assumed by all schemes, so we don't list it explicitly here."
+trust comment: "Very high because of minimal assumptions: Only relies on some rather weak properties of the hash function used (collision resistance is only required for the hash function used on the message itself)"
+sources:
+  - RFC8391 https://tools.ietf.org/html/rfc8391
```
Comment (franziskuskiefer) [inline signatures/xmss/xmss.yaml]: The NIST draft for SP 800-208 should be referenced as well.
https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-208-draft.pdf

File: schema/paramset.json (hunk 2 above)
Comment (dingens) [reply schema/paramset.json]: Agreed. I just pushed them to master directly, I assume they'll disappear from here automatically, let's see

File: schema/flavor.json (hunk 1 above)
Comment (dingens) [reply schema/flavor.json]: No, the line I added is the EUF-CMA one. (I hate JSON for not allowing trailing commas grml)

File: signatures/xmss/xmss-sha2/param/xmss-sha2-10-256.yaml (hunk 5 above)
Comment (dingens) [reply signatures/xmss/xmss-sha2/param/xmss-sha2-10-256.yaml:4]: Of course, my bad. I even found an explicit source for it (RFC § 5)

File: schema/flavor.json (hunk 1 above)
Comment (franziskuskiefer) [reply schema/flavor.json:27]: this shouldn't be necessary anymore (and not in this patch anyway :slightly_smiling_face: )

**Comment by dingens** [comment]
> lgtm without the `flavor.json` changes

they are now gone (except a tab to space change). can you have a quick look and merge then?

(I'll add xmss-mt later separately.)

**Pull request merged**